File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:

        subsets = []
        for r in range(len(nums) + 1):
            subsets.extend(itertools.combinations(nums, r))

        total=0
        for i in subsets:
            sum = 0
            for j in i :
                sum ^=j
            total+=sum
            
        return total

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.java
class Solution {

    public static int find( int n,int xor,int[] arr ) { 
        
        //base case
        if( n ==arr.length ) return xor;

        int take = find( n+1,xor^arr[n],arr );
        int ntake = find( n+1,xor,arr );

        return take+ntake;
    }
    public int subsetXORSum(int[] nums) {
        
        return find( 0,0,nums );
    }
}